contrib/internal/httptrace/httptrace.go

```diff
@@ -55,3 +81,69 @@ func FinishRequestSpan(s tracer.Span, status int, opts ...tracer.FinishOption) {
 	}
 	s.Finish(opts...)
 }
+
+// Helper function to return the IP network out of a string.
+func ippref(s string) *netaddr.IPPrefix {
+	if prefix, err := netaddr.ParseIPPrefix(s); err == nil {
+		return &prefix
+	}
+	return nil
+}
+
+// getClientIP uses the request headers to resolve the client IP. If a specific header to check is provided through
+// DD_CLIENT_IP_HEADER, then only this header is checked.
+func getClientIP(remoteAddr string, headers http.Header, clientIPHeader string) netaddr.IP {
```

Comment:
```suggestion
func getClientIP(req *http.Request) netaddr.IP {
```
You shouldn't need to pass two arguments. Frankly, you shouldn't even need to pass the second argument because it is globally accessible and you can replace it in tests if you want to.